### PR TITLE
 Updated to snake cases, solved bugs and Enhancement

### PR DESCRIPTION
**Pull request by MUTHU-RAKESH-27**
Network Settings Intent - Fixed issue with the message of the day key in the code, fixed the snmp and syslog servers updation issue and Removing the servers.
Device Credential Intent - Camel Case to Snake Case
Template Intent - Camel Case to Snake Case, Skeleton for Export and Import Project/Template.
Playbooks - Updated Playbooks for all the above modules.
dnac.py - Updated check_task_response_status() function

### CODE REVIEW

File: plugins/modules/network_settings_intent.py [hunk 1]
```diff
@@ -785,15 +785,16 @@ def get_network_params(self, site_id):
...
-                    "port": netflow_details.get("value")[0].get("port"),
-                    "configureDnacIP": netflow_details.get("value")[0].get("configureDnacIP"),
+                    "port": netflow_details.get("value")[0].get("port")
                 },
                 "timezone": timezone_details.get("value")[0],
             }
         }
         network_settings = network_details.get("settings")
-        if dhcp_details is not None:
+        if dhcp_details.get("value") != []:
```
Comment (madhansansel) [inline plugins/modules/network_settings_intent.py:794]: As we already got the value of dhcp_details by using "dhcp_details.get("value")", do we need to get again in next line? Can we optimize it by storing and reusing it..

```
dhcp_value = dhcp_details.get("value")
if dhcp_value:
    network_settings.update({"dhcpServer": dhcp_value})
else:
    network_settings.update({"dhcpServer": [""]})
```
(or)
As we call the same API "network_settings.update" in if and else, we can write below:
```
dhcp_value = dhcp_details.get("value")
network_settings.update({"dhcpServer": dhcp_value if dhcp_value else [""]})
```

File: plugins/modules/network_settings_intent.py [hunk 2]
```diff
@@ -804,35 +805,66 @@ def get_network_params(self, site_id):
                 }
             })
 
-        if ntpserver_details is not None:
+        if ntpserver_details.get("value") != []:
```
Comment (madhansansel) [inline plugins/modules/network_settings_intent.py:808]: Same as like dhcp_value above

File: plugins/modules/network_settings_intent.py [hunk 3]
```diff
@@ -804,35 +805,66 @@ def get_network_params(self, site_id):
...
 
         if messageoftheday_details is not None:
             network_settings.update({
                 "messageOfTheday": {
                     "bannerMessage": messageoftheday_details.get("value")[0].get("bannerMessage"),
-                    "retainExistingBanner":
-                    messageoftheday_details.get("value")[0].get("retainExistingBanner"),
                 }
             })
+            if messageoftheday_details.get("value")[0].get("retainExistingBanner") is True:
```
Comment (madhansansel) [inline plugins/modules/network_settings_intent.py:819]: same as like above..
```
retain_existing_banner = messageoftheday_details.get("value")[0].get("retainExistingBanner")
if retain_existing_banner is True:
    network_settings.get("messageOfTheDay").update({
        "retainExistingBanner": "true"
    })
else:
    network_settings.get("messageOfTheDay").update({
        "retainExistingBanner": "false"
    })
```

(or)
```
retain_existing_banner = messageoftheday_details.get("value")[0].get("retainExistingBanner")
network_settings["messageOfTheDay"]["retainExistingBanner"] = str(retain_existing_banner).lower()
```

File: plugins/modules/network_settings_intent.py [hunk 4]
```diff
@@ -804,35 +805,66 @@ def get_network_params(self, site_id):
...
 
         if network_aaa and network_aaa_pan:
-            network_settings.update({
-                "network_aaa": {
-                    "network": network_aaa.get("value")[0].get("ipAddress"),
-                    "protocol": network_aaa.get("value")[0].get("protocol"),
-                    "ipAddress": network_aaa_pan.get("value")[0]
-                }
-            })
+            if network_aaa_pan.get("value")[0] == "None":
```
Comment (madhansansel) [inline plugins/modules/network_settings_intent.py:829]: Same like above
```
aaa_pan_value = network_aaa_pan.get("value")[0]
aaa_value = network_aaa.get("value")[0]

network_settings.update({
    "network_aaa": {
        "network": aaa_value.get("ipAddress"),
        "protocol": aaa_value.get("protocol"),
        "ipAddress": aaa_pan_value if aaa_pan_value != "None" else aaa_value.get("ipAddress"),
        "servers": "AAA" if aaa_pan_value == "None" else "ISE"
    }
})

```

(0r)
```
aaa_pan_value = network_aaa_pan.get("value")[0]
aaa_value = network_aaa.get("value")[0]

if aaa_pan_value == "None":
    network_settings.update({
        "network_aaa": {
            "network": aaa_value.get("ipAddress"),
            "protocol": aaa_value.get("protocol"),
            "ipAddress": network_aaa2.get("value")[0].get("ipAddress"),
            "servers": "AAA"
        }
    })
else:
    network_settings.update({
        "network_aaa": {
            "network": aaa_value.get("ipAddress"),
            "protocol": aaa_value.get("protocol"),
            "ipAddress": aaa_pan_value,
            "servers": "ISE"
        }
    })

```

File: plugins/modules/network_settings_intent.py [hunk 5]
```diff
@@ -804,35 +805,66 @@ def get_network_params(self, site_id):
...
 
         if clientAndEndpoint_aaa and clientAndEndpoint_aaa_pan:
-            network_settings.update({
-                "clientAndEndpoint_aaa": {
-                    "network": clientAndEndpoint_aaa.get("value")[0].get("ipAddress"),
-                    "protocol": clientAndEndpoint_aaa.get("value")[0].get("protocol"),
-                    "ipAddress": clientAndEndpoint_aaa_pan.get("value")[0],
-                }
-            })
+            if clientAndEndpoint_aaa_pan.get("value")[0] == "None":
```
Comment (madhansansel) [inline plugins/modules/network_settings_intent.py:849]: Same like above, we are getting the value  clientAndEndpoint_aaa_pan.get("value")[0]  4 times in both if and else part

File: plugins/modules/template_intent.py [hunk 6]
```diff
@@ -265,282 +525,575 @@
...
-            type: bool
-          selection:
-            description: Configuration Template Create's selection.
+          validation_errors:
+            description: Configuration Template Create's validationErrors.
             suboptions:
-              defaultSelectedValues:
-                description: Default selection values.
-                elements: str
+              rollbackTemplateErrors:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:532]: are we still using camecase?

File: plugins/modules/template_intent.py [hunk 7]
```diff
@@ -265,282 +525,575 @@
...
+                description: Validation or design conflicts errors of rollback template.
+                elements: dict
                 type: list
-              id:
-                description: UUID of selection.
+              templateErrors:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:536]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 8]
```diff
@@ -265,282 +525,575 @@
...
+                description: Validation or design conflicts errors.
+                elements: dict
+                type: list
+              templateId:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:540]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 9]
```diff
@@ -265,282 +525,575 @@
...
+                description: UUID of template.
                 type: str
-              selectionType:
-                description: Type of selection(SINGLE_SELECT or MULTI_SELECT).
+              templateVersion:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:543]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 10]
```diff
@@ -265,282 +525,575 @@
...
-        description: Current version of template.
-        type: str
-      versionDescription:
-        description: Template version comments.
-        type: str
+          project:
+            description: Import the project details.
+            type: dict
+            suboptions:
+              doVersion:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:580]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 11]
```diff
@@ -265,282 +525,575 @@
...
+                description: DoVersion query parameter. If this flag is true, creates a new
+                  version of the template with the imported contents, if the templates already
+                  exists. " If false and if template already exists, then operation
+                  fails with 'Template already exists' error.
+                type: bool
+          template:
+            description: Import the template details.
+            type: dict
+            suboptions:
+              doVersion:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:590]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 12]
```diff
@@ -265,282 +525,575 @@
...
+                description: Configuration Template Import Template's payload.
+                elements: dict
+                suboptions:
+                  author:
+                    description: Author of template.
+                    type: str
+                  composite:
+                    description: Is it composite template.
+                    type: bool
+                  containingTemplates:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:606]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 13]
```diff
@@ -265,282 +525,575 @@
...
+                        description: Description of template.
+                        type: str
+                      device_types:
+                        description: Configuration Template Import Template's deviceTypes.
+                        elements: dict
+                        suboptions:
+                          productFamily:
+                            description: Device family.
+                            type: str
+                          productSeries:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:623]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 14]
```diff
@@ -265,282 +525,575 @@
...
+                          id:
+                            description: UUID of template param.
+                            type: str
+                          instructionText:
+                            description: Instruction text for param.
+                            type: str
+                          key:
+                            description: Key.
+                            type: str
+                          notParam:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:676]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 15]
```diff
@@ -265,282 +525,575 @@
...
+                            description: Is it not a variable.
+                            type: bool
+                          order:
+                            description: Order of template param.
+                            type: int
+                          paramArray:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:682]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 16]
```diff
@@ -265,282 +525,575 @@
...
+                            description: Is it an array.
+                            type: bool
+                          parameterName:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:685]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 17]
```diff
@@ -265,282 +525,575 @@
...
+                                description: Min value of range.
+                                type: int
+                            type: list
+                          required:
+                            description: Is param required.
+                            type: bool
+                          selection:
+                            description: Configuration Template Import Template's selection.
+                            suboptions:
+                              defaultSelectedValues:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:711]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 18]
```diff
@@ -265,282 +525,575 @@
...
+                                description: Default selection values.
+                                elements: str
+                                type: list
+                              id:
+                                description: UUID of selection.
+                                type: str
+                              selectionType:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:718]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 19]
```diff
@@ -265,282 +525,575 @@
...
+                                description: Type of selection(SINGLE_SELECT or MULTI_SELECT).
+                                type: str
+                              selectionValues:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:721]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 20]
```diff
@@ -265,282 +525,575 @@
...
+                            description: UUID of tag.
+                            type: str
+                          name:
+                            description: Name of tag.
+                            type: str
+                        type: list
+                      template_content:
+                        description: Template content.
+                        type: str
+                      templateParams:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:740]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 21]
```diff
@@ -265,282 +525,575 @@
...
+                        description: Configuration Template Import Template's templateParams.
+                        elements: dict
+                        suboptions:
+                          binding:
+                            description: Bind to source.
+                            type: str
+                          customOrder:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:747]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 22]
```diff
@@ -265,282 +525,575 @@
...
+                            description: CustomOrder of template param.
+                            type: int
+                          dataType:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:750]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 23]
```diff
@@ -265,282 +525,575 @@
...
+                            description: Datatype of template param.
+                            type: str
+                          defaultValue:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:753]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 24]
```diff
@@ -265,282 +525,575 @@
...
+                            description: Default value of template param.
+                            type: str
+                          description:
+                            description: Description of template param.
+                            type: str
+                          displayName:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:759]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 25]
```diff
@@ -265,282 +525,575 @@
...
+                            description: Display name of param.
+                            type: str
+                          group:
+                            description: Group.
+                            type: str
+                          id:
+                            description: UUID of template param.
+                            type: str
+                          instructionText:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:768]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 26]
```diff
@@ -265,282 +525,575 @@
...
+                            description: Instruction text for param.
+                            type: str
+                          key:
+                            description: Key.
+                            type: str
+                          notParam:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:774]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 27]
```diff
@@ -265,282 +525,575 @@
...
+                            description: Is it not a variable.
+                            type: bool
+                          order:
+                            description: Order of template param.
+                            type: int
+                          paramArray:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:780]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 28]
```diff
@@ -265,282 +525,575 @@
...
+                            description: Is it an array.
+                            type: bool
+                          parameterName:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:783]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 29]
```diff
@@ -265,282 +525,575 @@
...
+                            description: Provider.
+                            type: str
+                          range:
+                            description: Configuration Template Import Template's range.
+                            elements: dict
+                            suboptions:
+                              id:
+                                description: UUID of range.
+                                type: str
+                              maxValue:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:796]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 30]
```diff
@@ -265,282 +525,575 @@
...
+                                description: Max value of range.
+                                type: int
+                              minValue:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:799]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 31]
```diff
@@ -265,282 +525,575 @@
...
+                                description: Min value of range.
+                                type: int
+                            type: list
+                          required:
+                            description: Is param required.
+                            type: bool
+                          selection:
+                            description: Configuration Template Import Template's selection.
+                            suboptions:
+                              defaultSelectedValues:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:809]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 32]
```diff
@@ -265,282 +525,575 @@
...
+                                description: Default selection values.
+                                elements: str
+                                type: list
+                              id:
+                                description: UUID of selection.
+                                type: str
+                              selectionType:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:816]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 33]
```diff
@@ -265,282 +525,575 @@
...
+                                description: Type of selection(SINGLE_SELECT or MULTI_SELECT).
+                                type: str
+                              selectionValues:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:819]: are we still using camelcase?

File: plugins/modules/template_intent.py [hunk 34]
```diff
@@ -265,282 +525,575 @@
...
+                                description: Selection values.
+                                type: dict
+                            type: dict
+                        type: list
+                      version:
+                        description: Current version of template.
+                        type: str
+                    type: list
+                  createTime:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:828]: are we still using camelcase? can you please check the entire code?

File: plugins/modules/template_intent.py (hunk 6 above)
Comment (MUTHU-RAKESH-27) [reply plugins/modules/template_intent.py:532]: Done

File: plugins/modules/template_intent.py (hunk 22 above)
Comment (MUTHU-RAKESH-27) [reply plugins/modules/template_intent.py:750]: Export and import template and projects are blocked so we are using camel case similar to one to one API.

File: plugins/modules/template_intent.py (hunk 23 above)
Comment (MUTHU-RAKESH-27) [reply plugins/modules/template_intent.py:753]: Export and import template and projects are blocked so we are using camel case similar to one to one API.

File: plugins/modules/template_intent.py (hunk 25 above)
Comment (MUTHU-RAKESH-27) [reply plugins/modules/template_intent.py:768]: Export and import template and projects are blocked so we are using camel case similar to one to one API.

File: plugins/modules/template_intent.py (hunk 27 above)
Comment (MUTHU-RAKESH-27) [reply plugins/modules/template_intent.py:780]: Export and import template and projects are blocked so we are using camel case similar to one to one API.

File: plugins/modules/template_intent.py (hunk 24 above)
Comment (MUTHU-RAKESH-27) [reply plugins/modules/template_intent.py:759]: Export and import template and projects are blocked so we are using camel case similar to one to one API.

File: plugins/modules/template_intent.py (hunk 28 above)
Comment (MUTHU-RAKESH-27) [reply plugins/modules/template_intent.py:783]: Export and import template and projects are blocked so we are using camel case similar to one to one API.

File: plugins/modules/template_intent.py (hunk 21 above)
Comment (MUTHU-RAKESH-27) [reply plugins/modules/template_intent.py:747]: Export and import template and projects are blocked so we are using camel case similar to one to one API.

File: plugins/modules/template_intent.py (hunk 16 above)
Comment (MUTHU-RAKESH-27) [reply plugins/modules/template_intent.py:685]: Export and import template and projects are blocked so we are using camel case similar to one to one API.

**Pull request closed without merging**